core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2597,14 +2597,30 @@ private[spark] object Utils extends Logging {
   }
 
   /**
-   * Push based shuffle can only be enabled when the application is submitted
-   * to run in YARN mode, with external shuffle service enabled
+   * Push based shuffle can only be enabled when below conditions are met:
+   *   - the application is submitted to run in YARN mode
+   *   - external shuffle service enabled
+   *   - IO encryption disabled
+   *   - serializer(such as KryoSerializer) supports relocation of serialized objects
```

Comment:
Do we have another available relocation-supporting serializer except `KryoSerializer`?

Reply:
No. According to [Tuning Spark - Data Serialization](https://spark.apache.org/docs/latest/tuning.html#data-serialization), `KryoSerializer` and `JavaSerializer` are the only available options for data serialization, `KryoSerializer` is relocation-supporting while `JavaSerializer` is not.
[UnsafeRowSerializer](https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/sql/core/src/main/scala/org/apache/spark/sql/execution/UnsafeRowSerializer.scala#L48) is relocation-supporting, but it seems only used for serializing UnsafeRows during shuffle.

Reply:
Ya, I thought like that. Thanks.